examples/with-apollo/lib/withData.js

```diff
@@ -0,0 +1,44 @@
+import { ApolloProvider, getDataFromTree } from 'react-apollo'
+import React from 'react'
+import 'isomorphic-fetch'
+import { IS_SERVER } from './exenv'
+import { initClient } from './initClient'
+import { initStore } from './initStore'
+
+export default (Component) => (
+  class extends React.Component {
+    static async getInitialProps (ctx) {
+      const headers = ctx.req ? ctx.req.headers : {}
+      const client = initClient(headers)
+      const store = initStore(client, client.initialState)
+
+      if (IS_SERVER) {
+        const app = (
+          <ApolloProvider client={client} store={store}>
+            <Component url={{ query: ctx.query, pathname: ctx.pathname }} />
+          </ApolloProvider>
+        )
+        await getDataFromTree(app)
+      }
+
+      return {
+        initialState: store.getState(),
```

Comment:
@ads1018 here you go: https://github.com/ads1018/next-apollo-example/compare/develop...tpreusse:broken-query-id?expand=1